types/atom/linter/index.d.ts

```diff
@@ -73,10 +75,80 @@ export interface IndieDelegate {
     dispose(): void;
 }
 
+export type LintResult = Array<Message|LegacyMessage> | null;
+
 export interface LinterProvider {
     name: string;
     scope: "file"|"project";
     lintsOnChange: boolean;
     grammarScopes: string[];
-    lint(textEditor: TextEditor): Message[]|null|Promise<Message[]|null>;
+    lint(editor: TextEditor): LintResult | Promise<LintResult>;
```

Comment:
No, you can't return a legacy message from V2 provider and expect it to work, it will break horribly. Believe me, I've tried because Markdown stuff is pointless bull IMO.

Reply:
That's surprising because the documentation states that you can. I don't think people will generally use v1 messages with the v2 provider, so I'll remove this.

Reply:
Maybe I'm being wrong and confusing something with something else. Let me check.

Reply:
Nope, pretty sure I'm right, at least judging by the quick look at the source code https://github.com/steelbrain/linter/blob/7d7e52625da118f51df73b5083d751a047effd44/lib/linter-registry.js#L126 -- linter version is being determined once on creation.